node/node.go

```diff
@@ -0,0 +1,452 @@
+	if err != nil {
+		return nil, nil, err
+	} else if err := cm.ConsensusSetSubscribe(ws, ccid, nil); err != nil {
+		return nil, nil, err
+	}
+	w := wallet.NewSingleAddressWallet(walletKey, ws)
+
+	dbDir := filepath.Join(dir, "db")
+	if err := os.MkdirAll(dbDir, 0700); err != nil {
+		return nil, nil, err
+	}
+	dbConn := stores.NewSQLiteConnection(filepath.Join(dbDir, "db.sqlite"))
+
+	hostdbDir := filepath.Join(dir, "hostdb")
+	if err := os.MkdirAll(hostdbDir, 0700); err != nil {
+		return nil, nil, err
+	}
+	sqlStore, ccid, err := stores.NewSQLStore(dbConn, true)
+	if err != nil {
+		return nil, nil, err
+	} else if err := cm.ConsensusSetSubscribe(sqlStore, ccid, nil); err != nil {
+		return nil, nil, err
+	}
+
+	contractsDir := filepath.Join(dir, "contracts")
+	if err := os.MkdirAll(contractsDir, 0700); err != nil {
+		return nil, nil, err
+	}
+
+	objectsDir := filepath.Join(dir, "objects")
+	if err := os.MkdirAll(objectsDir, 0700); err != nil {
+		return nil, nil, err
+	}
+
+	cleanup := func() error {
+		errs := []error{
+			g.Close(),
+			cm.Close(),
+			tp.Close(),
+			sqlStore.Close(),
+		}
+		for _, err := range errs {
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	b := bus.New(syncer{g, tp}, chainManager{cm}, txpool{tp}, w, sqlStore, sqlStore, sqlStore, sqlStore)
+	return b, cleanup, nil
+}
+
+func newWorker(cfg WorkerConfig, walletKey consensus.PrivateKey) (*worker.Worker, func() error, error) {
+	b := bus.NewClient(cfg.BusAddr, cfg.BusPassword)
+	workerKey := blake2b.Sum256(append([]byte("worker"), walletKey...))
+	w := worker.New(workerKey, b)
+	return w, func() error { return nil }, nil
+}
+
+func newAutopilot(cfg AutopilotConfig, dir string) (*autopilot.Autopilot, func() error, error) {
+	store, err := stores.NewJSONAutopilotStore(dir)
+	if err != nil {
+		return nil, nil, err
+	}
+	b := bus.NewClient(cfg.BusAddr, cfg.BusPassword)
+	w := worker.NewClient(cfg.WorkerAddr, cfg.WorkerPassword)
+	a, err := autopilot.New(store, b, w, cfg.LoopInterval)
+	if err != nil {
+		return nil, nil, err
+	}
+	cleanup := func() error {
+		a.Stop()
+		return nil
+	}
+	return a, cleanup, nil
+}
+
+// Node describes a single renterd process with one or more enabled components.
+// A node always needs to either create a bus or connect to an existing one by
+// calling CreateBus or AddBus respectively.
+// Afterwards a worker can be created or added with CreateWorker/AddWorker. If
+// the worker is created, it will automatically connect to the known bus. If it
+// is added, the caller needs to guarantee that the worker is connected to the
+// same bus as specified in the node.
+// After both a bus and a worker were added, the autopilot can be created. It
+// will connect to the bus and worker from the previous steps.
+type Node struct {
+	busAddr     *string
+	busPassword *string
+	bus         *bus.Bus
+
+	workerAddr     *string
+	workerPassword *string
+	w              *worker.Worker
```

Comment:
multi-worker support might be low hanging fruit and make the MVP way cooler

Reply:
I don't think it's necessary for the MVP  because we already got multi-worker support. Just not for the autopilot. Although maybe that's not true since we don't have EA uploads which might be a requirement for true multi-worker support. 

My guess is that multiple workers in the autopilot is also not going to change the `Node` type. Cause we would probably add additional workers to the autopilot through the autopilot's API instead of calling `node.AddWorker`. So the autopilot internally tracks the number of workers it is connected to and the one in the `Node` is merely the initial one.